mapclientplugins/geometricfitstep/view/geometricfitwidget.py

```diff
@@ -453,207 +482,287 @@ def _displaySurfacesTranslucentClicked(self):
-        if group:
-            checkBoxDisable = False
-            groupName = group.getName()
-            dataWeight, isLocallySet, inheritable = config.getGroupDataWeight(groupName)
-            dataWeightStr = realFormat.format(dataWeight)
-            #print("group:", group.getName(), "Weight", dataWeight, isLocallySet,inheritable)
-            if inheritable:
-                checkBoxTristate = True
-                if isLocallySet is not None:
-                    checkBoxState = QtCore.Qt.PartiallyChecked
-            if isLocallySet:
-                checkBoxState = QtCore.Qt.Checked
-                lineEditDisable = False
-        self._ui.configDataWeight_checkBox.setDisabled(checkBoxDisable)
+        checkBoxTristate,checkBoxState,lineEditDisable,dataWeightStr = self._getGroupSettingDisplayState(self._getFit().getGroupDataWeight)
         self._ui.configDataWeight_checkBox.setTristate(checkBoxTristate)
         self._ui.configDataWeight_checkBox.setCheckState(checkBoxState)
         self._ui.configDataWeight_lineEdit.setDisabled(lineEditDisable)
         self._ui.configDataWeight_lineEdit.setText(dataWeightStr)
 
     def _configDataWeightClicked(self):
         checkState = self._ui.configDataWeight_checkBox.checkState()
-        group = self._ui.configSettingGroup_fieldChooser.getField().getName()
-        self._ui.configDataWeight_lineEdit.setDisabled(True)
+        group = self._getGroup()
         if checkState == QtCore.Qt.Unchecked:
-            self._getConfig().setGroupDataWeight(group, None)
+            self._getFit().setGroupDataWeight(group, None)
         elif checkState == QtCore.Qt.PartiallyChecked:
-            self._getConfig().clearGroupDataWeight(group)
+            self._getFit().clearGroupDataWeight(group)
         else:
-            self._ui.configDataWeight_lineEdit.setDisabled(False)
             self._configDataWeightEntered()
         self._updateDataWeight()
 
     def _configDataWeightEntered(self):
         value = QLineEdit_parseRealNonNegative(self._ui.configDataWeight_lineEdit)
-        group = self._ui.configSettingGroup_fieldChooser.getField().getName()
+        group = self._getGroup()
         if value > 0.0:
-            self._getConfig().setGroupDataWeight(group, value)
+            self._getFit().setGroupDataWeight(group, value)
         else:
             print("Invalid model Data Weight entered")
         self._updateDataWeight()
 
+    def _updateStrainPenalty(self):
+        checkBoxTristate,checkBoxState,lineEditDisable,dataStr = self._getGroupSettingDisplayState(self._getFit().getGroupStrainPenalty)
+        self._ui.configStrainPenalty_checkBox.setTristate(checkBoxTristate)
+        self._ui.configStrainPenalty_checkBox.setCheckState(checkBoxState)        
+        self._ui.configStrainPenalty_lineEdit.setDisabled(lineEditDisable)
+        self._ui.configStrainPenalty_lineEdit.setText(dataStr)
 
+    def _configStrainPenaltyClicked(self):
```

Comment:
As mentioned earlier for widget member variables, make sure you update functions to match their category, e.g. 
```
_updateGroupFitStrainPenalty
_groupFitStrainPenaltyClicked
_groupFitStrainPenaltyEntered
etc.
```